server/api.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+type API struct {
+	plugin *Plugin
+	router *mux.Router
+}
+
+func (a *API) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	a.router.ServeHTTP(w, r)
+}
+
+func (a *API) handlerRemoveAttachments(w http.ResponseWriter, r *http.Request) {
+	// Get post_id from the query parameters
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	postID := r.FormValue("post_id")
+
+	// Check if the user is the post author or a system admin
+	userID := r.Header.Get("Mattermost-User-ID")
+	post, appErr := a.plugin.API.GetPost(postID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	if errReason := a.plugin.userHasRemovePermissionsToPost(userID, post.ChannelId, postID); errReason != "" {
+		http.Error(w, errReason, http.StatusForbidden)
+		return
+	}
+
+	// If the message is empty just delete the post
+	if post.Message == "" {
```

Comment:
The customer has the setting enabled that prevents users from deleting posts.  
<img width="616" alt="image" src="https://github.com/user-attachments/assets/0751bcf5-886d-4d3c-a35b-66a16b0fbb76">

They also do not allow editing posts after minutes.  I think we should honour those settings.

Reply:
Not sure if `userHasRemovePermissionsToPost` or `DeletePost` respect those settings already.

Reply:
it honors the edit time, not the delete post. It would be weird because a post with no message will just be a name with nothing. Maybe I can put a message like `(attachments removed)` or something like that for empty messages?

Reply:
Ok I have removed the code block that delete posts entirely, rather than checking the permission as well to avoid more complexity.